tests/persistence/test_persistence_short.py

```diff
@@ -0,0 +1,780 @@
+        fee_close=fee.return_value,
+        exchange='kraken',
+        is_short=True,
+        leverage=5.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPER4
+    )
+
+    assert float(round(trade.calculate_interest(), 8)) == round(0.689938580475, 8)
+    trade.open_date = datetime.utcnow() - timedelta(hours=0, minutes=10)
+    assert float(round(trade.calculate_interest(interest_rate=0.00025), 8)
+                 ) == round(0.229979526825, 8)
+
+
+@ pytest.mark.usefixtures("init_persistence")
+def test_interest_binance_short(market_short_order, fee):
+    """
+        Market trade on Binance at 3x and 5x leverage
+        Short trade
+        interest_rate: 0.05%, 0.25% per 1 day
+        open_rate: 0.00004173 base
+        close_rate: 0.00004099 base
+        amount:
+            91.99181073 * leverage(3) = 275.97543219 crypto
+            91.99181073 * leverage(5) = 459.95905365 crypto
+        borrowed:
+            275.97543219  crypto
+            459.95905365  crypto
+        time-periods: 10 minutes(rounds up to 1/24 time-period of 1 day)
+                        5 hours = 5/24
+
+        interest: borrowed * interest_rate * time-periods
+                    = 275.97543219 * 0.0005 * 1/24 = 0.005749488170625 crypto
+                    = 275.97543219 * 0.00025 * 5/24 = 0.0143737204265625 crypto
+                    = 459.95905365 * 0.0005 * 5/24 = 0.047912401421875 crypto
+                    = 459.95905365 * 0.00025 * 1/24 = 0.0047912401421875 crypto
+    """
+
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.001,
+        amount=275.97543219,
+        open_rate=0.00001099,
+        open_date=datetime.utcnow() - timedelta(hours=0, minutes=10),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        is_short=True,
+        leverage=3.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+
+    assert float(round(trade.calculate_interest(), 8)) == 0.00574949
+    trade.open_date = datetime.utcnow() - timedelta(hours=4, minutes=55)
+    assert float(round(trade.calculate_interest(interest_rate=0.00025), 8)) == 0.01437372
+
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.001,
+        amount=459.95905365,
+        open_rate=0.00001099,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        is_short=True,
+        leverage=5.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+
+    assert float(round(trade.calculate_interest(), 8)) == 0.04791240
+    trade.open_date = datetime.utcnow() - timedelta(hours=0, minutes=10)
+    assert float(round(trade.calculate_interest(interest_rate=0.00025), 8)) == 0.00479124
+
+
+@ pytest.mark.usefixtures("init_persistence")
+def test_calc_open_trade_value_short(market_short_order, fee):
```

Comment:
Maybe parametrize this test (and check if that can remove a 2nd test).

what i'd like to see is the difference between "long" and "short" (that's what this test tests) - with the remaining params left identical (obviously you'll have 2 parameters, is_short and expected (1 and 2 - probably).

While i think both ways are tested in 2 seperate tests - combining and parametrizing that will make it obvious at a glance how we expect the result to differ.

Reply:
I did something similar to that